bentoml/adapters/annotated_image_input.py

```diff
@@ -0,0 +1,384 @@
+    """
+    _, extension = os.path.splitext(file_name)
+    if extension.lower() not in accept_format_list:
+        raise BadInput(
+            "Input file not in supported format list: {}".format(accept_format_list)
+        )
+
+
+class AnnotatedImageInput(BaseInputAdapter):
+    """Transform incoming image data from http request, cli or lambda event into a
+    numpy array, while allowing an optional JSON file for image annotations (such
+    as object bounding boxes, class labels, etc.)
+
+    Transforms input image file into a numpy array, and loads JSON file as
+    a JSON serializable Python object, providing them to user-defined
+    API functions.
+
+    Args:
+        image_input_name (string): An acceptable input name for HTTP request
+            and function keyword argument. Default value is "image"
+        annotation_input_name (string): An acceptable input name for HTTP request
+            and function keyword argument. Default value is "annotations"
+        accept_image_formats (string[]):  A list of acceptable image formats.
+            Default value is loaded from bentoml config
+            'apiserver/default_image_input_accept_file_extensions', which is
+            set to ['.jpg', '.png', '.jpeg', '.tiff', '.webp', '.bmp'] by default.
+            List of all supported format can be found here:
+            https://imageio.readthedocs.io/en/stable/formats.html
+        pilmode (string): The pilmode to be used for reading image file into numpy
+            array. Default value is 'RGB'.  Find more information at:
+            https://imageio.readthedocs.io/en/stable/format_png-pil.html
+
+    Raises:
+        ImportError: imageio package is required to use AnnotatedImageInput
+
+    Example:
+
+        >>> from bentoml import BentoService, api, artifacts
+        >>> from bentoml.artifact import TensorflowArtifact
+        >>> from bentoml.adapters import AnnotatedImageInput
+        >>>
+        >>> CLASS_NAMES = ['cat', 'dog']
+        >>>
+        >>> @artifacts([TensorflowArtifact('classifer')])
+        >>> class PetClassification(BentoService):
+        >>>    @api(input=AnnotatedImageInput())
+        >>>    def predict(self, image, annotations):
+        >>>        cropped_pets = some_pet_finder(image, annotations)
+        >>>        results = self.artifacts.classifer.predict(cropped_pets)
+        >>>        return [CLASS_NAMES[r] for r in results]
+    """
+
+    HTTP_METHODS = ["POST"]
+    BATCH_MODE_SUPPORTED = True
+
+    def __init__(
+        self,
+        accept_image_formats=None,
+        image_input_name="image",
+        annotation_input_name="annotations",
+        pilmode="RGB",
+        is_batch_input=False,
+        **base_kwargs,
+    ):
+        assert imageio, "`imageio` dependency can be imported"
+
+        super(AnnotatedImageInput, self).__init__(
+            is_batch_input=is_batch_input, **base_kwargs
+        )
+        if 'input_names' in base_kwargs:
```

Comment:
It is ok to remove l156-l160,  `input_names` check was in the ImageInput because of the older ImageHandler API supports multiple image input, which was moved to the MultiImageInput adapter, and it was meant for a warning for users using the older API.

Reply:
Removed in the latest version!